test/extension/mobileTargetManager.test.ts

```diff
@@ -404,25 +401,221 @@ suite("MobileTargetManager", function () {
                     ].filter(target => target.isOnline)
                 );
             });
+            getAvdNameById = Sinon.stub(
+                adbHelper,
+                "getAvdNameById",
+                async (targetId: string): Promise<string | undefined> => {
+                    return [
+                        onlineSimulator1,
+                        onlineSimulator2,
+                        offlineSimulator1,
+                        offlineSimulator2,
+                        device1,
+                        device2,
+                    ].find(target => target.id === targetId)?.name;
+                },
+            );
             targetsForSelection = [];
+        }
+
+        suiteSetup(() => {
+            revertTargetsStates = () => {
+                onlineSimulator1 = {
+                    name: "emulatorName1",
+                    id: "emulator-5551",
+                    isVirtualTarget: true,
+                    isOnline: true,
+                };
+                onlineSimulator2 = {
+                    name: "emulatorName2",
+                    id: "emulator-5552",
+                    isVirtualTarget: true,
+                    isOnline: true,
+                };
+
+                offlineSimulator1 = {
+                    name: "emulatorName3",
+                    id: undefined,
+                    isVirtualTarget: true,
+                    isOnline: false,
+                }; //id: emulator-5553
+                offlineSimulator2 = {
+                    name: "emulatorName4",
+                    id: undefined,
+                    isVirtualTarget: true,
+                    isOnline: false,
+                }; //id: emulator-5554
+
+                device1 = { id: "deviceid1", isVirtualTarget: false, isOnline: true };
+                device2 = { id: "deviceid2", isVirtualTarget: false, isOnline: true };
+            };
+            defaultSetup();
         });
 
         suiteTeardown(() => {
-            getAbdsNamesStub.reset();
+            getAvdsNamesStub.reset();
             getOnlineTargetsStub.reset();
+            getAvdNameById.reset();
             launchSimulatorStub.reset();
         });
 
         suite("Target selection", function () {
             setup(async () => {
+                revertTargetsStates();
                 await targetManager.collectTargets();
+                targetsForSelection = [];
             });
 
             runTargetSelectionTests();
         });
 
+        suite("Collect targets in case there in no 'emulator' utility in the PATH", function () {
```

Comment:
```suggestion
        suite("Collect targets in case there is no 'emulator' utility in the PATH", function () {
```
```suggestion
        suite("Collect targets in case there in no 'emulator' utility in the PATH", function () {
```